### ISSUE TITLE
Mount share as read-only to match shared folders privilage settings

### ISSUE THREAD

**Issue by 3ed** [OpenMediaVault-Plugin-Developers/openmediavault-sftp#6]
command:
```
mount --bind -o ro /old /new
```
fstab:
```
/old /new none bind,ro 0 0
```

**Comment by ryecoaaron** [comment]
bind mounting as read/write shouldn't give a user any more permissions than they have on the source folder.

**Comment by 3ed** [comment]
But can give them less. In `Shared folder` is `Privileges` where we can set permission to share. Columns: name, read-write, read-only, no access. When:
- read-write - as now
- read-only - read-only bind
- no access - without bind

Like in ftp, samba, etc. But if I know correctly, sshd do not have application level permissions like eg. proftpd have.

**Comment by ryecoaaron** [comment]
I don't see the point of no access.  Don't associate the shared folder with that user if you don't want them to have access.  We looked at bind mounting read only but a read only bind mount doesn't respect the ro permissions.  Try it.

**Comment by 3ed** [comment]
Yes looks like I assumed that this works in debian jessie.
I shoud not done this test on arch instead of debian before I submit:
```
mkdir /tmp/root
mount --bind -o ro /root /tmp/root
LANGUAGE=C touch /tmp/root/1
touch: cannot touch '/tmp/root/1': Read-only file system
```
Sorry. Maybe debian9 has this issue fixed?

> I don't see the point of no access. Don't associate the shared folder with that user if you don't want them to have access.

This would be nice to have this like in ftp or to not have `Privileges` at all. But nevermind. :) This method not working.

**Comment by ryecoaaron** [comment]
It does seem to work on Debian 9.  I guess I can look at adding this.

**Comment by ryecoaaron** [comment]
Fix has been committed but I need to test.

**Comment by ryecoaaron** [comment]
Fixed in 3.1
